examples/custom-assessments/README.md

```diff
@@ -0,0 +1,62 @@
+# Custom Assessments on AML
+
+In this folder, we show how to run custom assessment episodes with a trained RL agent on Azure ML
+with a custom Gymnasium environment ("Simple Adder").
+
+### What this sample covers
+
+- How to evaluate the agent with custom episode configurations
+
+### What this sample does not cover
+
+- How to train an agent using the simulation environment on AML
+- How to optimize the training algorithm for best performance
+- How to visualize the performance of the agent
+- How to deploy the agent
+
+## Prerequisites
+
+- You have a trained agent available (for example, you followed our tutorial `getting-started-on-aml`)
+- You have the simulation environment used for training the agent available.
+- (Optional) To run assessments on AML instead of locally, you will need to install the [Azure CLI (with the ML extension)](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-configure-cli?view=azureml-api-2&tabs=public) and an [AML environment](https://azure.github.io/plato/#aml-environment-setup) with the same Python and package versions you used for training the agent.
+
+## Example Overview
+
+This example shows how to run custom assessments on a trained agent using Ray and AML. The example uses the "Simple Adder" simulation environment and a trained agent from the `getting-started-on-aml` example. The simulation environment is modified to enable custom episode configurations via the _options_ parameter in [gymnasium.Env.reset()](https://gymnasium.farama.org/api/env/#gymnasium.Env.reset). Ray is then used to run parallel assessments on different episode configurations, either locally or on AML. The assessment results are saved to a CSV file for further analysis and visualization.
+
+## Tutorial
+
+1. Add the simulation environment to `./scr/sim.py` and modify the `reset()` method to pass episode configurations via the _options_ input parameter.
+
+2. Add the episode configurations to `./init_conditions.json`.
+
+3. Download the agent checkpoints locally to the ``./checkpoints`` folder.
+
+4. Activate the Python environment you used to train the agent.
+
+5. To test locally, run the following command in the `./src` folder:
+
+```bash
+python main.py --test-local
+```
+WARNING: Running custom assessments locally may overload your machine depending on various factors, such as the specifications of your machine, the simulation environment complexity, the assessment size and duration, the data and model size, etc. You can adjust the number of Ray workers or run the code on AML to avoid this.
+
+6. (Optional) To run on AML, follow these steps:
```

Comment:
There was an issue with job getting stuck, maybe worth adding additional logs to troubleshoot issues?

Reply:
I was able to reproduce this issue. The reason it hangs is due to the num_rollout_workers defined during training (in your case, 10 workers). RLlib's `Algorithm.from_checkpoint` uses the same config from training (and again allocates 10 workers). However, since we are now using the Ray Remote API instead of rllib, we do not use the rllib workers. Therefore, I changed the main.py code to use `restore` instead of `from_checkpoint` in order to set num_rollout_workers=0 before the checkpoint is restored.